sky/shell/platform/android/io/flutter/view/FlutterMain.java

```diff
@@ -113,24 +164,31 @@ public static void startInitialization(Context applicationContext) {
 
     /**
      * Blocks until initialization of the native system has completed.
-     **/
+     */
     public static void ensureInitializationComplete(Context applicationContext, String[] args) {
         if (sInitialized) {
             return;
         }
         try {
             sResourceExtractor.waitForCompletion();
 
-            String[] shellArgs = (args != null) ? Arrays.copyOf(args, args.length + 1) : new String[1];
+            List<String> shellArgs = new ArrayList<>();
+            if (args != null) {
+                Collections.addAll(shellArgs, args);
+            }
             if (sIsPrecompiled) {
-                shellArgs[shellArgs.length - 1] =
-                    "--aot-snapshot-path=" + PathUtils.getDataDirectory(applicationContext);
+                shellArgs.add("--" + AOT_SNAPSHOT_PATH_KEY + "=" +
+                    PathUtils.getDataDirectory(applicationContext));
```

Comment:
Oh, I see.  We pass the base directory and the file names separately.  I guess that's ok.